bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java

```diff
@@ -337,8 +340,14 @@ private BigtableOptions(
     if (!Strings.isNullOrEmpty(projectId)
         && !Strings.isNullOrEmpty(instanceId)) {
       this.instanceName = new BigtableInstanceName(projectId, instanceId);
+      if (!Strings.isNullOrEmpty(clusterId)) {
+        this.clusterName = instanceName.toClusterName(clusterId);
+      } else {
+        this.clusterName = null;
+      }
     } else {
       this.instanceName = null;
+      this.clusterName = null;
```

Comment:
nit: explicitly set/default them to null first

this.instanceName = null;
this.clusterName = null;

if (...)
  this.instanceName = ... 
  if (...)
    this.clusterName = ...